src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -833,8 +855,9 @@ sync(const std::string& port, xclBOSyncDirection dir, size_t sz, size_t offset)
 xrtBufferHandle
 xrtBOAllocUserPtr(xrtDeviceHandle dhdl, void* userptr, size_t size, xrtBufferFlags flags, xrtMemoryGroup grp)
 {
+  NATIVE_LOG_FUNCTION_CALL ;
   try {
-    auto boh = alloc(get_xcl_device_handle(dhdl), userptr, size, flags, grp);
+    auto boh = alloc_userptr(get_xcl_device_handle(dhdl), userptr, size, flags, grp);
     bo_cache[boh.get()] = boh;
     return boh.get();
   }
```

Comment:
Let use the wrapper here also.   Use the lambda function aka what I proposed for get_size().

Reply:
Replaced all C-API logging with calls through the wrapper